### ISSUE TITLE
Hetero NN: why output-data does not report all the expected output ?

### ISSUE THREAD

**Issue by amyseoj1** [FederatedAI/FATE#5012]
Hello, 
I am working with a binary data, and 
my models are defined as 

### Guest Bottom, Top Model
guest_bottom = t.nn.Sequential(
    nn.Linear(10, 6),	   # (column size, label size)
    nn.ReLU()
)
guest_top = t.nn.Sequential(
    nn.Linear(6, **2**), #output prediction and **variance**
    nn.Sigmoid()
)

### Host Bottom Model
host_bottom = t.nn.Sequential(
    nn.Linear(20, 6),               # (column size, label size)
    nn.ReLU()
)

I have a customised loss function to learn the variance in the data. 
The guest_top model has two output nodes: one for prediction probability and one for the variance. 

It seems like the output data is irrelevant to the output node size and only gives me probabilities for class 0 and class 1. 
No matter how many output nodes selected, the output data only reports probabilities for class 0 and 1. 

Could someone please help me out ?

**Comment by Yonggie** [comment]
Your top model outputs prediction in the shape of [6,2], and it'll be recognized as binary classification output, so you always get 0 and 1 probabilities.

BTW, but might not be relevent, the ``get_output_data`` function can only return the pre-defined data, such as logits, binary, etc..
To return the customized results we can rewrite the ``predict`` function in FedAvg Trainer in ``fate_project_path/federatedml/nn/homo/trainer/fedavg_trainer.py``, and save the prediction there, without ``get output data``. It works when I encounter similar problem but in homo-fed learning.  There might be similar way in hetero scenario.

**Comment by amyseoj1** [comment]
Hello Yonggie, 

Could you provide me a sample code ? I don't quite get "save the prediction there" .
Thank you for your reply !

**Comment by Yonggie** [comment]
Hi Amy,

It means to write ``torch.save(certain_path)`` in the ``predict`` function. After training you will find the "customized output" in ``certain_path``. For the record, I did it in homo fed scenario.
